x-pack/plugins/fleet/server/routes/agent_policy/handlers.ts

```diff
@@ -153,6 +185,33 @@ export const createAgentPolicyHandler: FleetRequestHandler<
   }
 };
 
+async function createPackagePolicy(
+  soClient: SavedObjectsClientContract,
+  esClient: ElasticsearchClient,
+  agentPolicy: AgentPolicy,
+  packageToInstall: string,
+  options: { spaceId: string; user: AuthenticatedUser | undefined }
+) {
+  // If needed, retrieve package information and build a new package policy for the package
+  // NOTE: we ignore failures in attempting to create package policy, since agent policy might have been created
+  // successfully
+  const newPackagePolicy = await packagePolicyService
+    .buildPackagePolicyFromPackage(soClient, packageToInstall)
+    .catch(() => undefined);
+
+  if (!newPackagePolicy) return;
+
+  newPackagePolicy.policy_id = agentPolicy.id;
+  newPackagePolicy.namespace = agentPolicy.namespace;
+  newPackagePolicy.name = await incrementPackageName(soClient, packageToInstall);
```

Comment:
It feels like we should be able to move this increment logic into the packagePolicyService somehow so we don't have to handle it in multiple places.

Reply:
it is already imported from packagePolicyService
`import { incrementPackageName } from '../../services/package_policy';`

there is another logic on UI to increment agent policy name, however that is needed there as we want to show the incremented name to users before submitting to backend

Reply:
it is already imported from packagePolicyService
`import { incrementPackageName } from '../../services/package_policy';`

there is another logic on UI to increment agent policy name, however that is needed there as we want to show the incremented name to users before submitting to backend